### PR TITLE
Specify Manure Type in C&S ManureNutrient Request

### LINKED ISSUE #1044: Specify Manure Type in Soil and Crop Manure Application Requests
**Is your feature request related to a problem? Please describe.**
The type of Manure (i.e. solid or liquid) is an important factor for consideration in Manure application. Currently the manure application is based on N & P alone. The form of manure (specifically, how much water is associated with the manure nutrients and solids) is an important factor to consider in manure application and in soil nutrient cycling. Thus, we need to be able to specify the form of the manure when the Soil and Crop module requests manure from the Manure Module. 

**Describe the solution you'd like**
I would like to add a variable that indicates the manure type to the Manure-Field connector request from the soil and crop module so that the Manure Module can only provide the nutrients from the relevant manure streams for application. 

**Describe alternatives you've considered**
An alternative would be to specify a range of values of acceptable Total Solid fractions but I think this is too specific and will be more difficult to track manure masses in the Manure Module

**Additional context**

### PR DESCRIPTION
**Pull request by ew3361zh**
## Update: 1071 has been merged into main and all the code from main is updated in this PR. I've removed the "Stacked" label accordingly.

PR to address issue #1044 to specify the manure type in c&s manure nutrient requests to the manure module.

To address this, the following changes were made:

In field.py:
1. Manure type added to `_execute_manure_application()` method.

In events.py:
1. Manure type added to `init` method.

In field_manager.py:
1. Manure type added to `_setup_manure_schedule()` method.
2. Manure type string from input files is converted to ManureType object here. Another location possibility I looked at was `_validate_manure_parameters` but that doesn't return anything and updating these as a side effect isn't a good practice.
3. I considered raising an error here but I believe the Enum class has its own set of built in Value and Type errors for these situations so not sure it's worth raising a separate error. We also have a check in IM's metadata to ensure the input data files with manure types are valid ManureType values.

In manure_schedule.py:
1. Manure types added to init, `generate_manure_events()`, and `_validate_manure_parameters()` methods.
2. Validation added for manure types (though not sure this is entirely necessary but I thought I'd follow the protocol for the other manure params.

Other updates:
1. After discussion with @Isgotkowitz, all the metadata and manure_schedule input files were updates with manure types.
2. Unit testing fixed to account for manure type being added throughout.
3. Tested successfully with simulations for all 7 metadata files.

### CODE REVIEW

File: input/metadata/ARL_metadata.json [hunk 1]
```diff
@@ -1976,6 +1976,15 @@
           "maximum": 1.0
         }
       },
+      "manure_types": {
+        "type": "array",
+        "description": "The type of manure which will be requested for the application.",
+        "default": [],
+        "properties": {
+          "type": "string",
+          "pattern": "(liquid|solid)"
```
Comment (Isgotkowitz) [inline input/metadata/ARL_metadata.json]:
```suggestion
          "pattern": "^(liquid|solid)$"
```

File: input/metadata/barnyard_metadata.json [hunk 2]
```diff
@@ -1990,6 +1990,15 @@
           "maximum": 1.0
         }
       },
+      "manure_types": {
+        "type": "array",
+        "description": "The type of manure which will be requested for the application.",
+        "default": [],
+        "properties": {
+          "type": "string",
+          "pattern": "(liquid|solid)"
```
Comment (Isgotkowitz) [inline input/metadata/barnyard_metadata.json]:
```suggestion
          "pattern": "^(liquid|solid)$"
```

File: input/metadata/default_metadata.json [hunk 3]
```diff
@@ -1983,6 +1983,15 @@
           "maximum": 1.0
         }
       },
+      "manure_types": {
+        "type": "array",
+        "description": "The type of manure which will be requested for the application.",
+        "default": [],
+        "properties": {
+          "type": "string",
+          "pattern": "(liquid|solid)"
```
Comment (Isgotkowitz) [inline input/metadata/default_metadata.json]:
```suggestion
          "pattern": "^(liquid|solid)$"
```

File: input/metadata/multi_crop_metadata.json [hunk 4]
```diff
@@ -1976,6 +1976,15 @@
           "maximum": 1.0
         }
       },
+      "manure_types": {
+        "type": "array",
+        "description": "The type of manure which will be requested for the application.",
+        "default": [],
+        "properties": {
+          "type": "string",
+          "pattern": "(liquid|solid)"
```
Comment (Isgotkowitz) [inline input/metadata/multi_crop_metadata.json]:
```suggestion
          "pattern": "^(liquid|solid)$"
```

File: input/metadata/swat_metadata.json [hunk 5]
```diff
@@ -1976,6 +1976,15 @@
           "maximum": 1.0
         }
       },
+      "manure_types": {
+        "type": "array",
+        "description": "The type of manure which will be requested for the application.",
+        "default": [],
+        "properties": {
+          "type": "string",
+          "pattern": "(liquid|solid)"
```
Comment (Isgotkowitz) [inline input/metadata/swat_metadata.json]:
```suggestion
          "pattern": "^(liquid|solid)$"
```

File: input/metadata/testing_metadata.json [hunk 6]
```diff
@@ -1976,6 +1976,15 @@
           "maximum": 1.0
         }
       },
+      "manure_types": {
+        "type": "array",
+        "description": "The type of manure which will be requested for the application.",
+        "default": [],
+        "properties": {
+          "type": "string",
+          "pattern": "(liquid|solid)"
```
Comment (Isgotkowitz) [inline input/metadata/testing_metadata.json]:
```suggestion
          "pattern": "^(liquid|solid)$"
```

File: input/metadata/user_defined_ration.json [hunk 7]
```diff
@@ -1983,6 +1983,15 @@
           "maximum": 1.0
         }
       },
+      "manure_types": {
+        "type": "array",
+        "description": "The type of manure which will be requested for the application.",
+        "default": [],
+        "properties": {
+          "type": "string",
+          "pattern": "(liquid|solid)"
```
Comment (Isgotkowitz) [inline input/metadata/user_defined_ration.json]:
```suggestion
          "pattern": "^(liquid|solid)$"
```

**Review by Isgotkowitz — APPROVED** [review]
Only suggestions are making the string patterns in the metadata more precise.

Nice job Niko!

**Review by elle-andreen — APPROVED** [review]
All looks good to me! The necessary instances for specification/alteration by manure type appear to all be updated here.

**Comment by ew3361zh** [comment]
Spoke with @KFosterReed this morning and she was good with merging this request with the current reviews/approvals.

**Pull request merged**